src/ServiceControl/CompositeViews/Messages/ScatterGatherApi.cs

```diff
@@ -21,32 +21,22 @@ public abstract class ScatterGatherApiBase
     {
     }
 
-    public record ScatterGatherContext(PagingInfo PagingInfo);
-
-    public abstract class ScatterGatherApi<TDataStore, TIn, TOut> : ScatterGatherApiBase, IApi
+    public abstract class ScatterGatherApi<TIn, TOut> : ScatterGatherApiBase, IApi
         where TIn : ScatterGatherContext
         where TOut : class
     {
-        protected ScatterGatherApi(TDataStore store, Settings settings, IHttpClientFactory httpClientFactory)
+        protected ScatterGatherApi(Settings settings, IHttpClientFactory httpClientFactory)
         {
-            DataStore = store;
             Settings = settings;
             HttpClientFactory = httpClientFactory;
             logger = LogManager.GetLogger(GetType());
         }
+        protected Settings Settings { get; }
+        protected IHttpClientFactory HttpClientFactory { get; }
 
-        protected TDataStore DataStore { get; }
-        Settings Settings { get; }
-        IHttpClientFactory HttpClientFactory { get; }
-
-        public async Task<QueryResult<TOut>> Execute(TIn input, string pathAndQuery)
+        protected void ExecuteRemotes(List<Task<QueryResult<TOut>>> tasks, string pathAndQuery)
```

Comment:
@bording, The only thing I dislike about the proposed changes is this `ExecuteRemotes` method. Looking at the API from the user perspective, it is not clear what the List should be.

Another option would be to design this like:

```suggestion
        protected List<Task<QueryResult<TOut>>> ExecuteRemotes(string pathAndQuery)
```

And then create the list instance inside the method. That was my original design, which comes with the downside of not being able to initialize the list with the right size.